core/include/kernel/dt_driver.h

```diff
@@ -60,4 +61,39 @@ struct dt_driver_provider {
 
 SLIST_HEAD(dt_driver_prov_list, dt_driver_provider);
 extern struct dt_driver_prov_list dt_driver_provider_list;
+
+/**
+ * dt_driver_register_provider - Register a driver provider
+ *
+ * @fdt: Device tree to work on
+ * @nodeoffset: Node offset of the clock
+ * @get_of_device: Callback to match the devicetree with a device instance
+ * @data: Data which will be passed to the @get_of_device callback
+ * @type: Driver type
+ *
+ * @get_of_device callback return a void *. Driver provider is expected to
+ * include a shim helper to cast to device reference into provider driver
+ * target structure reference (i.e. (struct clk *) for clock devices.
```

Comment:
Missing closing parenthese.
`i.e.` means "that is", not "for example" which would be better in this context. Do you mean `e.g.`?

Reply:
yes, i'll fix. thanks for the info :)